src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -2539,12 +2531,12 @@ def test_resource_policy_management_group(self, resource_group):
             self.resource_policy_operations(resource_group, management_group_name)
 
             # Attempt to get a policy definition at an invalid management group scope
-            with self.assertRaises(SystemExit):
+            with self.assertRaises(IncorrectUsageError):
                 self.cmd(self.cmdstring('policy definition show -n "/providers/microsoft.management/managementgroups/myMg/providers/microsoft.authorization/missingsegment"'))
         finally:
             self.cmd('account management-group delete -n ' + management_group_name)
 
-    @record_only()
+    @live_only()
```

Comment:
Why change this test from `@record_only()` to `@live_only()`

Reply:
This test will create policy assignment on a specific subscription. Policy assignment uses --scope argument. The mock doesn't work when the subscription comes from --scope. 
This test wasn't run in either live or record before my change.